From f92da6993e45b28a7e0c82b1aed89aae95f72931 Mon Sep 17 00:00:00 2001
From: Jose Dapena Paz <jdapena@igalia.com>
Date: Wed, 17 Jun 2020 22:57:35 +0000
Subject: [PATCH] GCC: remove noexcept declaration in CrossVariantMojoAssociatedReceive/Remote

It breaks compilation of blink::WebServiceRegistrationObjectInfo that includes
WebServiceWorkerObjectInfo instance. This one uses default move constructor
without noexcept declaration, but has as members CrossVariantMojoAssociatedReceive
and CrossVariantMojoAssociatedRemote with noexcept declaration in move constructor.
declared noexcept.

As resolved in other places, noexcept is not really much useful with -fno-exception
and our codebase is not prepare to add it consistently. Clang will not report
exception declaration incompatibilities in case -fno-exception is used, but GCC
fails.

Bug: 819294
Change-Id: Id5c46c271a8a92352d769de9c3c263d3dce4c9c7
Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/2247791
Commit-Queue: Dave Tapuska <dtapuska@chromium.org>
Reviewed-by: Dave Tapuska <dtapuska@chromium.org>
Cr-Commit-Position: refs/heads/master@{#779526}
---

diff --git a/third_party/blink/public/platform/cross_variant_mojo_util.h b/third_party/blink/public/platform/cross_variant_mojo_util.h
index 92c69ce..9945e00 100644
--- a/third_party/blink/public/platform/cross_variant_mojo_util.h
+++ b/third_party/blink/public/platform/cross_variant_mojo_util.h
@@ -131,10 +131,10 @@
   CrossVariantMojoAssociatedReceiver() = default;
   ~CrossVariantMojoAssociatedReceiver() = default;
 
-  CrossVariantMojoAssociatedReceiver(
-      CrossVariantMojoAssociatedReceiver&&) noexcept = default;
+  CrossVariantMojoAssociatedReceiver(CrossVariantMojoAssociatedReceiver&&) =
+      default;
   CrossVariantMojoAssociatedReceiver& operator=(
-      CrossVariantMojoAssociatedReceiver&&) noexcept = default;
+      CrossVariantMojoAssociatedReceiver&&) = default;
 
   CrossVariantMojoAssociatedReceiver(
       const CrossVariantMojoAssociatedReceiver&) = delete;
@@ -166,10 +166,10 @@
   CrossVariantMojoAssociatedRemote() = default;
   ~CrossVariantMojoAssociatedRemote() = default;
 
-  CrossVariantMojoAssociatedRemote(
-      CrossVariantMojoAssociatedRemote&&) noexcept = default;
+  CrossVariantMojoAssociatedRemote(CrossVariantMojoAssociatedRemote&&) =
+      default;
   CrossVariantMojoAssociatedRemote& operator=(
-      CrossVariantMojoAssociatedRemote&&) noexcept = default;
+      CrossVariantMojoAssociatedRemote&&) = default;
 
   CrossVariantMojoAssociatedRemote(const CrossVariantMojoAssociatedRemote&) =
       delete;
